internal/venice-common/src/main/java/com/linkedin/venice/acl/handler/AbstractStoreAclHandler.java

```diff
@@ -95,115 +79,55 @@ public void channelRead0(ChannelHandlerContext ctx, HttpRequest req) throws SSLP
+    } catch (VeniceStoreIsMigratedException storeIsMigratedException) {
+      LOGGER.debug("Requested store is being migrated: {} requested {} {}", client, method, req.uri());
+      errorHandler.accept(HttpResponseStatus.MOVED_PERMANENTLY, storeIsMigratedException.getMessage());
     }
   }
 
-  @Override
-  public <ReqT, RespT> ServerCall.Listener<ReqT> interceptCall(
-      ServerCall<ReqT, RespT> call,
-      Metadata headers,
-      ServerCallHandler<ReqT, RespT> next) {
-    return new ForwardingServerCallListener.SimpleForwardingServerCallListener<ReqT>(next.startCall(call, headers)) {
-      @Override
-      public void onMessage(ReqT message) {
-        VeniceClientRequest request = (VeniceClientRequest) message;
-        // For now, GRPC only supports STORAGE query
-        String storeName = extractStoreName(request.getResourceName(), QueryAction.STORAGE);
-        String method = request.getMethod();
-
-        BiConsumer<Status, Metadata> grpcCloseConsumer = call::close;
-        BiConsumer<HttpResponseStatus, String> errorHandler = ((httpResponseStatus, s) -> grpcCloseConsumer
-            .accept(httpResponseStatusToGrpcStatus(httpResponseStatus, s), headers));
-
-        if (StringUtils.isEmpty(storeName) || StringUtils.isEmpty(method)) {
-          LOGGER.error("Invalid store name {} or method {}", storeName, method);
-          grpcCloseConsumer.accept(Status.INVALID_ARGUMENT.withDescription("Invalid request"), headers);
-          return;
-        }
-
-        try {
-          X509Certificate clientCert = extractGrpcClientCert(call);
-          String client = Objects.requireNonNull(call.getAttributes().get(Grpc.TRANSPORT_ATTR_REMOTE_ADDR)).toString();
-
-          if (VeniceSystemStoreUtils.isSystemStore(storeName)
-              || hasAccess(client, call.getAuthority(), clientCert, storeName, method, errorHandler)) {
-            LOGGER.info("Requested principal has access to resource. Processing request");
-            super.onMessage(message);
-          }
-        } catch (SSLPeerUnverifiedException e) {
-          LOGGER.error("Cannot verify the certificate.", e);
-          grpcCloseConsumer.accept(Status.UNAUTHENTICATED.withDescription("Invalid certificate"), headers);
-        } catch (VeniceException e) {
-          LOGGER.error("Cannot process request successfully due to", e);
-          grpcCloseConsumer.accept(Status.INTERNAL.withDescription(e.getMessage()), headers);
-        }
-      }
-    };
-  }
-
-  /**
-   * Extract the store name from the incoming resource name.
-   */
-  protected String extractStoreName(String resourceName, QueryAction queryAction) {
-    return resourceName;
+  protected boolean isAccessAlreadyApproved(ChannelHandlerContext ctx) {
+    return false;
   }
 
-  @VisibleForTesting
-  boolean isInvalidRequest(String[] requestParts) {
-    int partsLength = requestParts.length;
-    boolean invalidRequest = false;
+  protected abstract boolean needsAclValidation(String[] requestParts);
```

Comment:
Should we just make this method take in a `RouterResourceType`? Either way I think it'd be great if we can have a test where we pass in or construct requestParts for all enums in `RouterResourceType` to ensure we are handling things correctly and didn't miss anything.

Reply:
This piece of code is shared b/w Router and Server. So, this function cannot use `RouterResourceType`. But let me see if I can do something to reduce the string parsing.

Also, the tests for this class were somewhat tedious to automate. I tried improving, but that would end up overwriting all tests for this class. So, I avoided it. I can give it another shot